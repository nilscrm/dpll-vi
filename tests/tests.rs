use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
