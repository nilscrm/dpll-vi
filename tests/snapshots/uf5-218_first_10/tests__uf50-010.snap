---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = false
var 2 = true
var 3 = true
var 4 = false
var 5 = true
var 6 = true
var 7 = true
var 8 = true
var 10 = false
var 11 = true
var 12 = true
var 13 = true
var 14 = false
var 15 = true
var 16 = false
var 17 = false
var 18 = true
var 19 = true
var 20 = true
var 21 = false
var 22 = false
var 23 = true
var 24 = false
var 25 = true
var 26 = true
var 28 = false
var 29 = false
var 30 = true
var 31 = true
var 32 = false
var 33 = true
var 34 = false
var 35 = true
var 36 = false
var 37 = false
var 38 = false
var 39 = false
var 41 = false
var 42 = false
var 43 = false
var 44 = false
var 45 = false
var 46 = true
var 47 = true
var 48 = true
var 49 = true
var 50 = true

----- stderr -----

Interactions
  Total           18_086_051
    Depth          9_163_269
    Breadth                1
  Annihilate      10_182_819
  Commute             35_537
  Copy             1_764_872
  Erase            2_075_257
  Expand           1_423_711
  Call             1_649_803
  Branch             954_052

Memory
  Heap               617_216 B
  Allocated      400_923_616 B
  Freed          400_923_616 B

[Performance...]
